ludwig/trainers/trainer.py

```diff
@@ -884,8 +884,21 @@ def train(
                     if self.distributed.is_model_parallel():
                         # Assume the full weights cannot fit in memory on GPU
                         self.model = self.model.cpu()
-                    _, unexpected_keys = self.model.load_state_dict(state_dict, strict=False)
-                    assert unexpected_keys == [], f"Unexpected keys found in state dict: {unexpected_keys}"
+
+                    # For a full explanation of this 8-bit workaround, see https://github.com/ludwig-ai/ludwig/pull/3606
+                    # TODO (jeffkinnison): Investigate why `self.model` is seemingly not placed on GPU before this point
+                    if hasattr(self.model.config_obj, "quantization") and self.model.config_obj.quantization.bits == 8:
+                        if torch.cuda.is_available():
+                            self.model.model.cuda()
+                            self.model.model.cpu()
```

Comment:
Ooof, maybe one callout here might be that the .cuda() call is unique and overriden for the Linear8Bit layers which internally does some stuff for 8BitParameters?

Reply:
Yeah, it's not clear to me why we need to move to GPU then back to CPU like this. Comment would be great so I don't need to read the full PR description.

Reply:
Added a comment and removed the move to CPU. It turns out the model was on GPU all along: `self.model.device` reports that it is on CPU, but `self.model.model.device` and deeper modules in the model all report that that they are on GPU.